### PR TITLE
Fix: Make `nn.init.orthogonal_` no-op for empty input

### LINKED ISSUE #73503: `torch.nn.init.orthogonal_` does check tensor with zero rows
### 🐛 Describe the bug

`torch.nn.init.orthogonal_` does check tensor with zero dimension

```python
import torch
input = torch.rand([0, 1])
torch.nn.init.orthogonal_(input)
# ZeroDivisionError: integer division or modulo by zero
```

### Versions

pytorch: 1.10.1

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @pbelevich

### PR DESCRIPTION
**Pull request by jaketae**
Fixes #73503.

### CODE REVIEW

File: torch/nn/init.py [hunk 1]
```diff
@@ -466,6 +466,9 @@ def orthogonal_(tensor, gain=1):
         raise ValueError("Only tensors with 2 or more dimensions are supported")
 
     rows = tensor.size(0)
+    if rows == 0:
```
Comment (albanD) [inline torch/nn/init.py:469]: nit: could we use `tensor.numel()` here to also catch the cases where other dimensions are of size 0

Comment (jaketae) [reply torch/nn/init.py]: Hey @albanD, thanks for the review. I've updated the code; is this in line with what you had in mind?

Comment (albanD) [reply torch/nn/init.py]: Yes!

**Review by albanD — APPROVED** [review]
SGTM

**Pull request closed without merging**